cmake/benchmark.cmake

```diff
@@ -16,7 +16,7 @@ else(benchmark_FOUND)
   FetchContent_Declare(
     benchmark
     GIT_REPOSITORY https://github.com/google/benchmark
-    GIT_TAG v1.7.1
+    GIT_TAG db3e000c1e6675f70ae41895ca1e25b274e52554
```

Comment:
Using the latest commit seems a bit dangerous. Especially because this specific commit has tripped some UBSAN errors in their CI. I recommend using version `v1.8.2`

Reply:
Interesting. I did not check with sanitizers, I'll try adding them and see.

Reply:
I have added options for sanitizers and it worked out fine:
it compiled and all tests passed except the test_new which produces memory leaks on purpose.

Reply:
Correction: ASAN worked fine, UBSAN does indeed warn that something deep in std::regex maybe-uninitialized and then the system is configured to fail on such warnings :-(

Reply:
Everything compiles on Ubuntu with g++-11.4, but benchmark with UBSAN breaks on Debian with g++-13.2. So I think it is a bug in standard library of g++-13.